vm/actor/tests/common/mod.rs

```diff
@@ -28,6 +28,7 @@ pub struct MockRuntime<'a, BS: BlockStore> {
     pub id_addresses: HashMap<Address, Address>,
     pub actor_code_cids: HashMap<Address, Cid>,
     pub new_actor_addr: Option<Address>,
+    pub message: UnsignedMessage,
```

Comment:
So about this, since all you need from this is caller, receiver, value_received and is currently storing that data twice (see above) you either need to remove the three fields above and handle those through the message type or update the runtime interface to not be hard coded `UnsignedMessage` and instead be an interface which returns those fields.

I think it makes sense to just do the former and add a TODO since that change is a bit more involved (cc: @ec2)

Reply:
Yeah it makes sense. I was thinking we need to make a Message trait. In this case, I think this is the only way to make it work. The message() function returns a reference to a UnsignedMessage, so its not like we can construct the message using the caller, receiever, value_received inside of that function

Reply:
Just to be clear about that, it is a bit awkward to use, but only for the mock runtime. I'm not convinced that making it generic is objectively better, can just handle those things through an unsigned message in the mock rt since it's only used for testing, but maybe yeah a bit clunky still. 

Having the message trait makes the mockrt impl more straightforward, at the cost of the actual runtime being unnecessarily complicated

Reply:
Yeah. I was thinking about it from the perspective of the MockRt. I dont expect us to be building any more runtimes anyways.